chia/data_layer/s3_plugin_service.py

```diff
@@ -19,25 +20,39 @@
 log = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True)
+class StoreConfig:
+    id: bytes32
+    bucket: Optional[str]
+    urls: Set[str]
+
+    @classmethod
+    def unmarshal(cls, d: Dict[str, Any]) -> StoreConfig:
+        upload_bucket = d.get("upload_bucket", None)
+        if len(upload_bucket) == 0:
+            upload_bucket = None
+
+        return StoreConfig(bytes32.from_hexstr(d["store_id"]), upload_bucket, d.get("download_urls", set()))
+
+    def marshal(self) -> Dict[str, Any]:
+        return {"store_id": self.id.hex(), "upload_bucket": self.bucket, "download_urls": self.urls}
+
+
 class S3Plugin:
     boto_client: boto3.client
     port: int
     region: str
     aws_access_key_id: str
     aws_secret_access_key: str
-    store_ids: List[bytes32]
-    bukets: Dict[str, List[str]]
-    urls: List[str]
+    store_ids: List[StoreConfig]
```

Comment:
```suggestion
    stores: List[StoreConfig]
```

Reply:
i guess theres also a comment from Kyle which doesn't appear while i review for some reason

Reply:
changed